pkg/controller/util.go

```diff
@@ -604,3 +609,196 @@ func (c *CloneController) objFromKey(informer cache.SharedIndexInformer, key int
 	}
 	return obj, nil
 }
+
+// CreateUploadPod creates upload service pod manifest and sends to server
+func CreateUploadPod(client kubernetes.Interface, caKeyPair *triple.KeyPair, clientCACert *x509.Certificate, image, verbose, pullPolicy, name string, pvc *v1.PersistentVolumeClaim) (*v1.Pod, error) {
+	ns := pvc.Namespace
+	commonName := name + ".namespace"
+	secretName := name + "-server-tls"
+	owner := MakeOwnerReference(pvc)
+
+	_, err := GetOrCreateServerKeyPair(client, ns, secretName, caKeyPair, commonName, name, &owner, clientCACert)
```

Comment:
any chance we could use the same key pairs for every upload server pod?

Reply:
Would have to disable hostname validation in the client's tls config.  I don't think we can use a wildcard cert in this case.  Other than less stuff, any advantages to using just one cert?

Reply:
yeah, juggling of less stuff.

it doesn't seem like hostname validation gives us much here. The uploadserver endpoint either has a cert signed by our CA or not.